src/quo2/components/inputs/browser_input/view.cljs

```diff
@@ -0,0 +1,87 @@
+
+(defn clear-icon
+  [blur? override-theme]
+  (if blur?
+    (colors/theme-colors colors/neutral-80-opa-30 colors/white-opa-10 override-theme)
+    (colors/theme-colors colors/neutral-40 colors/neutral-60 override-theme)))
+
+(defn- clear-button
+  [{:keys [on-press blur? override-theme]}]
+  [rn/touchable-opacity
+   {:accessibility-label :browser-input-clear-button
+    :style               style/clear-icon-container
+    :on-press            on-press}
+   [icon/icon :i/clear
+    {:color (clear-icon blur? override-theme)
+     :size  20}]])
+
+(defn cursor-color
+  [customization-color override-theme]
+  (colors/theme-colors (colors/custom-color customization-color 50)
+                       (colors/custom-color customization-color 60)
+                       override-theme))
+
+(defn placeholder-color
+  [state blur? override-theme]
+  (cond
+    (and blur? (= state :active))
+    (colors/theme-colors colors/neutral-80-opa-20 colors/white-opa-20 override-theme)
+
+    blur? ; state is default
+    (colors/theme-colors colors/neutral-80-opa-40 colors/white-opa-30 override-theme)
+
+    (= state :active)
+    (colors/theme-colors colors/neutral-30 colors/neutral-60 override-theme)
+
+    :else ; Not blur and state is default
+    (colors/theme-colors colors/neutral-40 colors/neutral-50 override-theme)))
+
+(def ^:private props-to-remove
+  [:cursor-color :placeholder-text-color :editable :on-change-text :on-focus
+   :on-blur :on-clear :value :disabled? :blur? :customization-color :override-theme])
+
+(defn browser-input
+  [{:keys [value]}]
+  (let [state       (reagent/atom :default)
+        set-active  #(reset! state :active)
+        set-default #(reset! state :default)
+        use-value?  (boolean value)]
+    (fn [{:keys [value disabled? blur? on-change-text customization-color
+                 on-clear on-focus on-blur override-theme]
+          :or   {customization-color :blue}
+          :as   props}]
+      (let [clean-props (apply dissoc props props-to-remove)]
+        [rn/view {:style style/container}
+         [rn/text-input
+          (cond-> {:accessibility-label    :browser-input
+                   :style                  (style/input-container disabled?)
+                   :cursor-color           (cursor-color customization-color override-theme)
+                   :selection-color        (when platform/ios? 
+                                             (cursor-color customization-color override-theme))
+                   :placeholder-text-color (placeholder-color @state blur? override-theme)
+                   :editable               (not disabled?)
+                   :keyboard-appearance    (colors/theme-colors :light :dark override-theme)
+                   :on-change-text         (fn [new-text]
+                                             (when on-change-text
+                                               (on-change-text new-text))
+                                             (reagent/flush))
```

Comment:
could you pls elaborate on reagent/flush usage

Reply:
It was added to trigger an instant update of the UI with the `on-change` event. 

I have updated the PR and removed it, it would no longer be needed as the component is an uncontrolled input now.